clients/venice-thin-client/src/main/java/com/linkedin/venice/client/schema/RouterBackedSchemaReader.java

```diff
@@ -39,7 +39,7 @@ public class RouterBackedSchemaReader implements SchemaReader {
   private static final Logger LOGGER = LogManager.getLogger(RouterBackedSchemaReader.class);
   private static final ObjectMapper OBJECT_MAPPER = ObjectMapperFactory.getInstance();
   private final Optional<Schema> readerSchema;
-  private Schema keySchema;
+  private volatile Schema keySchema;
```

Comment:
why do need volatile here?

Reply:
`keySchema` had a warning of inconsistent synchronization
```
In class com.linkedin.venice.client.schema.RouterBackedSchemaReader
Field com.linkedin.venice.client.schema.RouterBackedSchemaReader.keySchema
Synchronized 57% of the time
Unsynchronized access at RouterBackedSchemaReader.java:[line 67]
Unsynchronized access at RouterBackedSchemaReader.java:[line 68]
Synchronized access at RouterBackedSchemaReader.java:[line 71]
Synchronized access at RouterBackedSchemaReader.java:[line 72]
Synchronized access at RouterBackedSchemaReader.java:[line 78]
Synchronized access at RouterBackedSchemaReader.java:[line 79]
```

but after reading the code one more time, the inconsistent sync happens at line 67 and 68 is expected.  `volatile` is a redundant fix. I can remove it

Reply:
I just recalled the reason is not the inconsistent synchronization. It's because of double-checked locking. See [this page](http://www.cs.umd.edu/~pugh/java/memoryModel/DoubleCheckedLocking.html) for details.

```
  public Schema getKeySchema() {
    if (keySchema != null) {
      return keySchema;
    }
   // threadB is waiting at here
    synchronized (this) {
      if (keySchema != null) {
        return keySchema;
      }
       // threadA is fetching the schema
      SchemaEntry keySchemaEntry = fetchKeySchema();
      if (keySchemaEntry == null) {
        throw new VeniceClientException("Key Schema of store: " + this.storeName + " doesn't exist");
      }
      keySchema = keySchemaEntry.getSchema();
      return keySchema;
    }
  }
```

As codes shown above, when `threadA` fetches the schema but update of `keySchema` isn't visible to `threadB` yet, `threadB` will still try to fetch the schema. Making the `keySchema` `volatile` can resolve this potential issue.